pkg/service/backup_scheduler.go

```diff
@@ -216,6 +223,15 @@ func (h *BackupHandler) runIncrementalBackup(now time.Time) {
 	h.updateIncrementalBackupState()
 }
 
+func (h *BackupHandler) deleteEmptyBackup(stats *shared.BackupStat) {
+	if stats == nil || !stats.HasStats {
+		return
+	}
+	if stats.RecordCount == 0 && stats.UDFFileCount == 0 && stats.SecondaryIndexCount == 0 {
```

Comment:
It would be better to have a method under the `BackupStat` to hold this logic, something like `NotEmpty() bool`